### PR TITLE
Ddr/soln

### PR DESCRIPTION
**Pull request by tildeee**

### CODE REVIEW

File: generator.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
+def is_vowel? letter
+  return ["a", "e", "i", "o", "u"].include?(letter)
+end
+
+def run_generator
+
+  puts "What is the first word to combine?"
+
+  a = gets.chomp
+
```
Comment (droberts-sea) [inline generator.rb]: Might make sense to separate getting user input and doing the work - this method ends up being pretty long. Maybe have them write a method that gets a word from the user?

File: generator.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+  b_array = b.split("")
+
+  b_vowel = ""
+
+  b_array.reverse.each do | c |
+    if ["a", "e", "i", "o", "u"].include?(c)
+      b_vowel = c
+    end
```
Comment (droberts-sea) [inline generator.rb]: Use `is_vowel?` here.

File: generator.rb [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  end
+
+  a_rev = a.reverse
+
+  first = a_vowel.empty? ? a : a_rev.slice(a_rev.index(a_vowel) + 1, a_rev.length).reverse
+
+  second = b_vowel.empty? ? b : b.slice(b.index(b_vowel), b.length)
```
Comment (droberts-sea) [inline generator.rb]: Do we expect the students to do this fancy slice math?

File: generator.rb [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+  puts "#{ a } + #{ b } is #{ first + second }"
+end
+
+should_run = true
+
+while (should_run)
```
Comment (droberts-sea) [inline generator.rb:54]: Commenting on code as code, this should be wrapped in:

```ruby
if __FILE__ == $0
  # do work
end
```

If they're reading it, they can google it to figure out what it means.

**Review by droberts-sea — CHANGES_REQUESTED** [review]

File: generator.rb (hunk 4 above)
Comment (tildeee) [reply generator.rb:54]: Hm, I actually don't know this very well, does this look like this?

```ruby
if __FILE__ == $0
  should_run = true

  while (should_run)
    run_generator
    puts "continue?"
    should_run = gets.chomp == "yes"
  end
end
```

Thank you @droberts-ada !

**Comment by tildeee** [comment]
Closing this to open another PR against master

**Pull request closed without merging**